internal/querycoord/query_coord.go

```diff
@@ -471,9 +434,11 @@ func (qc *QueryCoord) handleNodeEvent(ctx context.Context) {
 					log.Error("QueryCoord failed to register a QueryNode", zap.Int64("nodeID", serverID), zap.String("error info", err.Error()))
 					continue
 				}
-				if err := qc.allocateNode(serverID); err != nil {
-					log.Error("unable to allcoate node", zap.Int64("nodeID", serverID), zap.Error(err))
-				}
+				go func(serverID int64) {
+					if err := qc.allocateNode(serverID); err != nil {
+						log.Error("unable to allcoate node", zap.Int64("nodeID", serverID), zap.Error(err))
+					}
+				}(serverID)
 				qc.metricsCacheManager.InvalidateSystemInfoMetrics()
```

Comment:
I'm not sure what exactly L442 does, but this change makes the executing order uncertain, which might be a problem?

Also I'm curious about why `registerNode` failure doesn't need to call L442, and why L442 needs to be called over and over again.

Reply:
It's ok to make new querynode register not randomly.

L442 is related to the metrics cache, if the new node registered successfully, the online node list shall be changed so the cached metrics shall be invalidated.